package.json

```diff
@@ -132,21 +132,26 @@
     "husky": "^8.0.2",
     "jest": "^29.3.1",
     "lint-staged": "^15.2.2",
-    "react": "^18.0.1",
-    "react-dom": "^18.0.1",
+    "react": "19.0.0",
+    "react-dom": "19.0.0",
     "serve": "^14.1.2",
     "start-server-and-test": "^1.15.2",
     "surge": "^0.23.1"
   },
   "peerDependencies": {
-    "react": "^17 || ^18",
-    "react-dom": "^17 || ^18"
+    "react": "^17 || ^18 || ^19",
+    "react-dom": "^17 || ^18 || ^19"
   },
   "engines": {
     "node": ">=18.12.0"
   },
   "packageManager": "pnpm@8.15.3",
   "volta": {
     "node": "18.19.1"
+  },
+  "pnpm": {
+    "patchedDependencies": {
+      "react-use@17.4.0": "patches/react-use@17.4.0.patch"
+    }
```

Comment:
required to get around the new ref types. I could have casted but this felt cleaner? happy to go either way to land this PR

Reply:
update: deleted the package instead and inlined the utility hooks